ledger/complete/wal/checkpoint_v6_test.go

```diff
@@ -167,3 +219,130 @@ func TestWriteAndReadCheckpointV6(t *testing.T) {
 		requireTriesEqual(t, tries, decoded)
 	})
 }
+
+func TestWriteAndReadCheckpointV6MultipleTries(t *testing.T) {
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries := createMultipleRandomTries(t)
+		fileName := "checkpoint-multi-file"
+		logger := unittest.Logger()
+		require.NoErrorf(t, StoreCheckpointV6Concurrent(tries, dir, fileName, &logger), "fail to store checkpoint")
+		decoded, err := ReadCheckpointV6(dir, fileName, &logger)
+		require.NoErrorf(t, err, "fail to read checkpoint %v/%v", dir, fileName)
+		requireTriesEqual(t, tries, decoded)
+	})
+}
+
+// test running checkpointing twice will produce the same checkpoint file
+func TestCheckpointV6IsDeterminstic(t *testing.T) {
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries := createMultipleRandomTries(t)
+		logger := unittest.Logger()
+		require.NoErrorf(t, StoreCheckpointV6Concurrent(tries, dir, "checkpoint1", &logger), "fail to store checkpoint")
```

Comment:
Maybe we should run both function in two separate goroutines to rule out possible state sharing?